libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/ListItem.kt

```diff
@@ -0,0 +1,419 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.designsystem.theme.components
+
+import androidx.compose.foundation.clickable
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Share
+import androidx.compose.material3.ListItemColors
+import androidx.compose.material3.ListItemDefaults
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.LocalTextStyle
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import io.element.android.libraries.designsystem.preview.ElementThemedPreview
+import io.element.android.libraries.designsystem.preview.PreviewGroup
+import io.element.android.libraries.theme.ElementTheme
+
+// Designs: https://www.figma.com/file/G1xy0HDZKJf5TCRFmKb5d5/Compound-Android-Components?type=design&node-id=425%3A24208&mode=design&t=G5hCfkLB6GgXDuWe-1
+
+/**
+ * A List Item component to be used in lists and menus with simple layouts, matching the Material 3 guidelines.
+ * @param headlineContent The main content of the list item, usually a text.
+ * @param modifier The modifier to be applied to the list item.
+ * @param overlineContent The content to be displayed above the headline content.
+ * @param supportingContent The content to be displayed below the headline content.
+ * @param leadingContent The content to be displayed before the headline content.
+ * @param trailingContent The content to be displayed after the headline content.
+ * @param colors The colors to be used for the list item.
+ * @param tonalElevation The tonal elevation to be used for the list item.
+ * @param shadowElevation The shadow elevation to be used for the list item.
+ * @param isPrimaryAction Whether the list item is a primary action. This will change the color of the leading content to use primary tokens.
+ * @param isDestructive Whether the list item is a destructive action. This will change the color of the headline content to use critical tokens.
+ * @param enabled Whether the list item is enabled. When disabled, will change the color of the headline content and the leading content to use disabled tokens.
+ * @param onClick The callback to be called when the list item is clicked.
+ *
+ * **Note:** [isPrimaryAction], [isDestructive] and [enabled] should only be used in single line [ListItem] components,
```

Comment:
There is no notion of "single line ListItem" as of yet. So this doc line leaves a lot to the dev's interpretation.

Perhaps we should remove `isPrimaryAction` and `isDestructive` from here, create another composable which only has a single line of content and also an enum param which is something like "StandardAction, PrimaryAction, DesctructiveAction".
This new composable could still be called `ListItem` (because the param list will be a lot different) or it could be called `SingleLineListItem`.

Reply:
Actually, this sounds better than what was implemented on https://github.com/vector-im/element-x-android/pull/990/commits/53ebf11b6a791e315bac375a9f24bb92f22e5b04.

I still need to create a couple of PRs to replace usages of `ListItems` through the app and add Lists to dialogs. Do you mind if I add this change in one of this other PRs, or even a dedicated one for it?

Reply:
Not a problem, but we can also leave it as is if it's not a problem to use "destructive" or "primary" style when there is also a supporting content.